homeassistant/components/switch/xiaomi_miio.py

```diff
@@ -38,31 +38,59 @@
 @asyncio.coroutine
 def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     """Set up the switch from config."""
-    from mirobo import Plug, DeviceException
+    from miio import Device, DeviceException
 
     host = config.get(CONF_HOST)
     name = config.get(CONF_NAME)
     token = config.get(CONF_TOKEN)
 
     _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
 
+    devices = []
     try:
-        plug = Plug(host, token)
+        plug = Device(host, token)
         device_info = plug.info()
         _LOGGER.info("%s %s %s initialized",
-                     device_info.raw['model'],
-                     device_info.raw['fw_ver'],
-                     device_info.raw['hw_ver'])
-
-        xiaomi_plug_switch = XiaomiPlugSwitch(name, plug, device_info)
+                     device_info.model,
+                     device_info.firmware_version,
+                     device_info.hardware_version)
+
+        if device_info.model in ['chuangmi.plug.v1']:
+            from miio import PlugV1
+            plug = PlugV1(host, token)
+
+            # The device has two switchable channels (mains and a USB port).
+            # A switch device per channel will be created.
+            for channel_usb in [True, False]:
+                device = ChuangMiPlugV1Switch(
+                    name, plug, device_info, channel_usb)
+                devices.append(device)
+
+        elif device_info.model in ['qmi.powerstrip.v1',
+                                   'zimi.powerstrip.v2']:
+            from miio import Strip
+            plug = Strip(host, token)
+            device = XiaomiPowerStripSwitch(name, plug, device_info)
+            devices.append(device)
+        elif device_info.model in ['chuangmi.plug.m1',
+                                   'chuangmi.plug.v2']:
+            from miio import Plug
+            plug = Plug(host, token)
+            device = XiaomiPlugGenericSwitch(name, plug, device_info)
+            devices.append(device)
+        else:
```

Comment:
Just a side-note, I think we need to start looking into getting these devices auto-detected (through mDNS), then we can create devices automatically based on that information.

Furthermore we could add a configurator for the UI to allow user input the tokens without touching the configuration files.